dash_enterprise_auth/__init__.py

```diff
@@ -98,18 +99,19 @@ def create_logout_button(label="Logout", style=None):
     )
 
 
-def _raise_context_error():
+def is_jupyter_kernel():
+    """Check if the current environment is a Jupyter kernel."""
     try:
-        is_jupyter_kernel = (
-            get_ipython().__class__.__name__ == "ZMQInteractiveShell"  # type: ignore
-        )
+        return get_ipython().__class__.__name__ == "ZMQInteractiveShell"  # type: ignore
     except NameError:
-        is_jupyter_kernel = False
+        return False
 
+
+def _raise_context_error():
     raise RuntimeError(
-        "dash-enterprise-auth functions should be called in a flask request context or dash callback and will not run in a notebook cell.\n"
+        "Could not get user token from the Jupyter kernel. Try closing and re-opening your notebook. \n"
```

Comment:
I don't have the historical knowledge, but just based on the comment you are replacing...
Do we need to consider the case of a jupyter kernel that is _not_ workspaces? Was the old error message only meant to be seen on DE?
Perhaps we need to keep this old error message in case users try to use DEA in a local notebook.